common/src/main/java/org/keycloak/common/Profile.java

```diff
@@ -121,6 +121,8 @@ public enum Feature {
         PASSKEYS("Passkeys", Type.PREVIEW),
 
         CACHE_EMBEDDED_REMOTE_STORE("Support for remote-store in embedded Infinispan caches", Type.EXPERIMENTAL),
+
+        USER_EVENT_METRICS("Collect metrics based on user events", Type.PREVIEW),
```

Comment:
Do you also plan to collect admin event metrics with via the associated `MicrometerMetricsEventListener`?
If that's the plan, then I'd recommend to use a more generic name for the feature to enable usage metric collection like `USAGE_EVENT_METRICS` as this would allow us to also collect metrics based on admin events later, without having to introduce a new feature flag.
The feature specific property to enable user event metrics collection `--user-event-metrics-enabled=true` is IMHO already specific enough. This can later also be extended to include `--admin-event-metrics-enabled=true`.

Reply:
The user event metrics might become fully supported earlier than the admin event metrics, so I'd like to keep it as is which is fine-grained.